enterprise/server/test/integration/remote_bazel/remote_bazel_test.go

```diff
@@ -248,6 +249,13 @@ func runLocalServerAndExecutor(t *testing.T, githubToken string) (*rbetest.Env,
 	executors := env.AddExecutors(t, 1)
 	require.Equal(t, 1, len(executors))
 	flags.Set(t, "executor.enable_bare_runner", true)
+
+	t.Cleanup(func() {
+		for _, e := range executors {
+			env.RemoveExecutor(e)
+		}
```

Comment:
hm how does removing the executors fix the race condition? wonder if rbetest should do this automatically, since we already call ShutdownBuildBuddyServers automatically

Reply:
actually it looks like `NewRBETestEnv` does register some executor shutdown logic as a cleanup func already, but not exactly RemoveExecutor - any idea what the difference is that makes this code work?

Reply:
Hmm that's a good point I'd missed that. I'm wondering if it's because `executor.env.GetHealthChecker().WaitForGracefulShutdown()` was in the background instead of the foreground. Let me test that out

Reply:
Added more details to the commit description